### DISCUSSION TITLE
apt-get update can't resolve when trying to build/run using buildah inside a running docker container

### DISCUSSION THREAD

**Discussion by rfwhitehead** [containers/buildah#5706] [Q&A]
I am trying to build Docker images in a Jenkins job, Jenkins running as a Docker container. When trying to run apt-get update either in the Dockerfile or a running Ubuntu image inside a running Docker container, I get the following output:

`Ign:1 http://archive.ubuntu.com/ubuntu jammy InRelease
Ign:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
Ign:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
Ign:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
Ign:1 http://archive.ubuntu.com/ubuntu jammy InRelease
Ign:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
Ign:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
Ign:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
Ign:1 http://archive.ubuntu.com/ubuntu jammy InRelease
Ign:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
Ign:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
Ign:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
Err:1 http://archive.ubuntu.com/ubuntu jammy InRelease
  System error resolving 'archive.ubuntu.com:80' - getaddrinfo (22: Invalid argument)
Err:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
  System error resolving 'security.ubuntu.com:80' - getaddrinfo (22: Invalid argument)
Err:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
  System error resolving 'archive.ubuntu.com:80' - getaddrinfo (22: Invalid argument)
Err:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
  System error resolving 'archive.ubuntu.com:80' - getaddrinfo (22: Invalid argument)
Reading package lists...
W: Problem unlinking the file /var/lib/apt/lists/partial/.apt-acquire-privs-test.odY0hI - IsAccessibleBySandboxUser (22: Invalid argument)
W: Problem unlinking the file /var/lib/apt/lists/partial/.apt-acquire-privs-test.VGaW1Y - IsAccessibleBySandboxUser (22: Invalid argument)
W: Problem unlinking the file /var/lib/apt/lists/partial/.apt-acquire-privs-test.7s2FkD - IsAccessibleBySandboxUser (22: Invalid argument)
W: Problem unlinking the file /var/lib/apt/lists/partial/.apt-acquire-privs-test.1LwGmA - IsAccessibleBySandboxUser (22: Invalid argument)
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/jammy/InRelease  System error resolving 'archive.ubuntu.com:80' - getaddrinfo (22: Invalid argument)
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/jammy-updates/InRelease  System error resolving 'archive.ubuntu.com:80' - getaddrinfo (22: Invalid argument)
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/jammy-backports/InRelease  System error resolving 'archive.ubuntu.com:80' - getaddrinfo (22: Invalid argument)
W: Failed to fetch http://security.ubuntu.com/ubuntu/dists/jammy-security/InRelease  System error resolving 'security.ubuntu.com:80' - getaddrinfo (22: Invalid argument)
W: Some index files failed to download. They have been ignored, or old ones used instead.`

**Comment by nalind** [comment]
Are you using host networking from inside of the container?  If not, what does the network configuration look like from `buildah info`?  That error message from `getaddrinfo(3)` suggests that a call it's making into libc or the kernel returned an EINVAL, and without further explanation, that makes it difficult to guess what exactly is failing here.

**Reply by rfwhitehead** [reply]
The initial Docker container is run with a bridge network mode. 

I then grab a generic Ubuntu image 

`buildah from --name build docker.io/ubuntu:22.04)`

run it with bash 

`buildah run build -bash `

and then when trying to run 
`apt-get update`
I get all the Ign and System error resolving errors

**Reply by nalind** [reply]
If the image isn't already set up to do so, try adding a `--network=host` to the `buildah run` command.  The container already has networking set up the way you want, so having `buildah` set up a network for the command it's running doesn't really gain you anything.
